### PR TITLE
Add linter for serializable resource

### PR DESCRIPTION
**Pull request by bf4**
Based on:
https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/lint.rb
https://github.com/rails/rails/blob/4-2-stable/activemodel/test/cases/lint_test.rb

With reference to:
https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/serializers/json.rb
https://github.com/rails/rails/blob/4-2-stable/activemodel/lib/active_model/serialization.rb

### CODE REVIEW

**Comment by bf4** [comment]
Transient failure, see https://github.com/rails-api/active_model_serializers/issues/995

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Serializer
     autoload :Configuration
     autoload :ArraySerializer
     autoload :Adapter
+    autoload :Lint
```
Comment (joaomdmoura) [inline lib/active_model/serializer.rb]: why autoload ```Lint``` here?

Comment (bf4) [inline lib/active_model/serializer.rb]: Only because it's cheap.  I'd be perfectly happy to change it to a `require 'active_model/serializer/lint'`

Comment (joaomdmoura) [inline lib/active_model/serializer.rb]: no worries, got it :smile:

**Comment by joaomdmoura** [comment]
LGTM, it's quite a small PR but useful for ppl trying to write some kind os tests. I'm merging it.

**Pull request merged**